src/Lykke.Service.PayInternal.Services/BtcTransferRequestService.cs

```diff
@@ -167,75 +112,82 @@ public BtcTransferRequestService(ITransferRepository transferRepository, IWallet
         {
-            var result = new TransferDto
+            sources = sources.Where(s => s.Amount > 0).ToList();
+            OnchainResponse request = null;
+            var result = new TransactionRequest
             {
                 DestinationAddress = destination,
                 Amount = sources.Sum(s => s.Amount),
-                Currency = BitcoinAssert,
-                SourceAddresses = sources,
-                TransferStatus = TransferStatus.InProgress,
-                TransferStatusError = TransferStatusError.NotError
+                Currency = BitcoinAsset,
+                SourceAmounts = sources,
+                CountConfirm = 1
             };
+
             var store = new MultipleTransferRequest
             {
-                Asset = BitcoinAssert,
+                Asset = BitcoinAsset,
+
                 Destination = destination,
                 Sources = (from s in sources
                            select new ToOneAddress(s.SourceAddress, s.Amount)).ToList()
             };
-            var r = await _bitcointApiClient.ApiTransactionMultipletransferPostWithHttpMessagesAsync();
-            var resData = r?.Body as TransactionIdAndHashResponse;
-            if (resData?.Hash == null)
+
+
+            if (sources.Any())
+            {
+                request = await _bitcointApiClient.TransactionMultipleTransfer(null, store.Destination, store.Asset, 0, 0, store.Sources);
+            }
+
+
+            if (request == null || request.HasError)
             {
-                var errorCode = (r?.Body as ApiException)?.Error.Code;
-                var errorMessage = (r?.Body as ApiException)?.Error.Message;
-                if (resData == null && errorCode == "3")
+
+                if (request?.Error?.ErrorCode == ErrorCode.NotEnoughBitcoinAvailable)
                 {
+                    var errorCode = request.Error.Code;
+                    var errorMessage = request.Error.Message;
                     await _log.WriteWarningAsync(nameof(BtcTransferRequestService), nameof(CreateBtcTransfer), JsonConvert.SerializeObject(store), $"Invalid amount. Error on TransactionMultipletransfer: {errorMessage} ({errorCode})");
-                    result.TransferStatus = TransferStatus.Error;
-                    result.TransferStatusError = TransferStatusError.InvalidAmount;
-                    if (await _transferRepository.SaveAsync(result) == null)
-                    {
-                        result.TransferStatusError = TransferStatusError.InternalError;
-                    }
-                    return result;
+
+                    return new Tuple<ITransactionRequest, TransferStatusError>(result, TransferStatusError.InvalidAddress);
+
                 }
 
                 await _log.WriteWarningAsync(nameof(BtcTransferRequestService), nameof(CreateBtcTransfer), JsonConvert.SerializeObject(store), "Transaction not confirmed.");
```

Comment:
I'd prefer if you use ToJson extension method instead of direct call of JsonConvert.SerializeObject